core/src/main/scala/com/mesosphere/usi/core/matching/RangeResource.scala

```diff
@@ -0,0 +1,251 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.ResourceUtil
+import com.mesosphere.usi.core.matching.RangeResource._
+import com.mesosphere.usi.core.models.{ResourceMatchResult, ResourceRequirement, ResourceType}
+import com.mesosphere.usi.core.protos.ProtoBuilders
+import org.apache.mesos.v1.Protos
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+/**
+  * Represents requirement for Mesos resource of type Range.
+  * You can either request static values (by providing concrete value) or dynamic values (by providing 0).
+  * Dynamic values are then selected from the offered ranges. If random implementation is passed in, these ranges are randomized.
+  * For more information about randomization see [[lazyRandomValuesFromRanges()]]
+  *
+  * @param requestedValues values pod wants to consume on the given ranges
+  * @param resourceType name of resource (e.g. ports)
+  * @param random when requesting dynamic values, you can provide Random implementation if you want dynamic values to be randomized
+  */
+case class RangeResource(requestedValues: Seq[Int], resourceType: ResourceType, random: Random = Random)
```

Comment:
Range resource is type of resource defined http://mesos.apache.org/documentation/attributes-resources/ ... this is way to express you want part of that resource. Similarly as you do it with `ScalarResource` e.g. for CPUs.

There might be similar strategies how you want to take part of that range. Right now I just implemented list of values. That's the list of values you want to consume out of that range.

You need the type because Resource can represent various types (names) of resources. As scalar can be of type mem or cpus, Range is also generic and can be of any type. In the `RangeResource` object there is helper method for ports that will create the resource for you without specifying the type.

Reply:
maybe it should be actually named `RangeRequirment` instead of `RangeResource`? WDYT?